### PR TITLE
fix(argo-cd): Use correct protocols for Istio service discovery

### PR DESCRIPTION
**Pull request by pdrastil**
PR fixes invalid port names used by Istio for protocol selection

Ref: https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/templates/argocd-applicationset/service.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ metadata:
 spec:
   type: {{ .Values.applicationSet.service.type }}
   ports:
-  - name: {{ .Values.applicationSet.service.portName }}
+  - name: http-webhook
```
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-applicationset/service.yaml:21]: This was originally `webhook`

File: charts/argo-cd/templates/argocd-repo-server/service.yaml [hunk 2]
```diff
@@ -16,9 +16,9 @@ metadata:
   namespace: {{ .Release.Namespace | quote }}
 spec:
   ports:
-  - name: {{ .Values.repoServer.service.portName }}
+  - name: tcp-repo-server
```
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-repo-server/service.yaml:19]: This was orginally `https`

**Comment by mkilchhofer** [comment]
I am not sure whether hardcoding is a good idea. We had several PRs/issues in the past where customers want customizing. Some customers want the names to follow "X" while others want "Y". Therefore I bias to let it as-is. But lets hear other opinions from maintainers.

**Comment by pdrastil** [comment]
@mkilchhofer Ok I'll revert and fix just names in values

**Comment by pdrastil** [comment]
Fixed only in values.yaml

**Review by mbevc1 — APPROVED** [review]

**Review by yu-croco — APPROVED** [review]

**Pull request merged**